ansible_rulebook/job_template_runner.py

```
@@ -120,8 +135,27 @@ async def _get_page(self, href_slug: str, params: dict) -> dict:
+            raise ControllerApiException(
+                f"Controller returned invalid JSON response: {str(e)}"
+            )
         except aiohttp.ClientError as e:
+            if self._controller_available:
+                self._controller_available = False
+                logger.warning(
+                    "Controller unavailable (%s), " "requests will be retried",
+                    e,
+                )
             logger.error(CLIENT_CONNECT_ERROR_STRING, str(e))
             raise ControllerApiException(str(e))
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)